sklearn/linear_model/tests/test_ridge.py

```diff
@@ -1823,3 +1823,33 @@ def test_ridgecv_normalize_deprecated(Estimator):
         FutureWarning, match=r"Set parameter alphas to: original_alphas \* n_samples"
     ):
         estimator.fit(X, y)
+
+
+@pytest.mark.parametrize("solver", ["sparse_cg", "sag"])
+def test_ridge_sample_weights_dense_sparse(solver, global_random_seed):
```

Comment:
Could this be merged with `test_ridge_fit_intercept_sparse` (and maybe also `test_ridge_fit_intercept_sparse`)? (take the best part of both).
It seems kind of duplicate.

Reply:
Right, I merged them into a single test. It allowed to find out that lbfgs also had the same issue. I fixed it as well